exporter/azuremonitorexporter/logexporter_test.go

```diff
@@ -128,6 +129,50 @@ func TestExporterLogDataCallback(t *testing.T) {
 	mockTransportChannel.AssertNumberOfCalls(t, "Send", 3)
 }
 
+func TestLogRecordToEnvelopeResourceAttributes(t *testing.T) {
+	logRecord := getTestLogRecord(t, 1)
+	logPacker := getLogPacker()
+	resource := getResource()
+
+	envelope := logPacker.LogRecordToEnvelope(logRecord, resource, getScope())
+
+	require.NotEmpty(t, resource.Attributes())
+	envelopeData := envelope.Data.(*contracts.Data).BaseData.(*contracts.MessageData)
+	require.Subset(t, envelopeData.Properties, resource.Attributes().AsRaw())
+}
+
+func TestLogRecordToEnvelopeInstrumentationScope(t *testing.T) {
+	const aiInstrumentationLibraryNameConvention = "instrumentationlibrary.name"
+	const aiInstrumentationLibraryVersionConvention = "instrumentationlibrary.version"
+
+	logRecord := getTestLogRecord(t, 1)
+	logPacker := getLogPacker()
+	scope := getScope()
+
+	envelope := logPacker.LogRecordToEnvelope(logRecord, getResource(), scope)
+
+	envelopeData := envelope.Data.(*contracts.Data).BaseData.(*contracts.MessageData)
+	require.Equal(t, scope.Name(), envelopeData.Properties[aiInstrumentationLibraryNameConvention])
+	require.Equal(t, scope.Version(), envelopeData.Properties[aiInstrumentationLibraryVersionConvention])
+}
+
+func TestLogRecordToEnvelopeCloudTags(t *testing.T) {
+	const aiCloudRoleConvention = "ai.cloud.role"
+	const aiCloudRoleInstanceConvention = "ai.cloud.roleInstance"
+
+	logRecord := getTestLogRecord(t, 1)
+	logPacker := getLogPacker()
+	resource := getResource()
+
+	envelope := logPacker.LogRecordToEnvelope(logRecord, resource, getScope())
+
+	resourceAttributes := resource.Attributes().AsRaw()
+	expectedCloudRole := resourceAttributes[conventions.AttributeServiceNamespace].(string) + "." + resourceAttributes[conventions.AttributeServiceName].(string)
+	require.Equal(t, expectedCloudRole, envelope.Tags[aiCloudRoleConvention])
+	expectedCloudRoleInstance := resourceAttributes[conventions.AttributeServiceInstanceID]
+	require.Equal(t, expectedCloudRoleInstance, envelope.Tags[aiCloudRoleInstanceConvention])
```

Comment:
I think it would make sense to use the same attribute constants that the production code uses here as well.

Reply:
Based on the discussion on the instrumentation scope constants, I'm okay leaving this as-is.